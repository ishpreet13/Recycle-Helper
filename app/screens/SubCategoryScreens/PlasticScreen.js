import React, { useEffect } from "react";
import { View, StyleSheet, Text } from "react-native";
import { BackHandler } from "react-native";

import AppButton from "../../components/AppButton";
import Screen from "../../components/Screen";
import Styles from "../../config/Styles";
import colors from "../../config/colors";

function PlasticScreen({ navigation }) {
  const handleBackButtonClick = () => {
    navigation.navigate("MainCategoryList");
    return true;
  };
  useEffect(() => {
    BackHandler.addEventListener("hardwareBackPress", handleBackButtonClick);
    return () => {
      BackHandler.removeEventListener(
        "hardwareBackPress",
        handleBackButtonClick
      );
    };
  }, []);
  return (
    <Screen>
      <View style={styles.container}>
        <View style={styles.heading}>
          <Text style={Styles.heading}>
            Select Plastic Category for Recycle Instructions
          </Text>
        </View>
        <View style={styles.buttonContainer}>
          <AppButton
            title="Packing Peanuts"
            onPress={() =>
              navigation.navigate("InstructionScreen", {
                documentName: "Packing Peanuts",
              })
            }
          ></AppButton>
          <AppButton
            title="Plastic Containers"
            onPress={() =>
              navigation.navigate("InstructionScreen", {
                documentName: "Plastic Container",
              })
            }
          ></AppButton>
          <AppButton
            title="Plastic Bags"
            onPress={() =>
              navigation.navigate("InstructionScreen", {
                documentName: "Plastic Bags",
              })
            }
          ></AppButton>
          <AppButton
            title="Plastic Cap and Lids"
            onPress={() =>
              navigation.navigate("InstructionScreen", {
                documentName: "Plastic Cap and Lids",
              })
            }
          ></AppButton>
          <AppButton
            title="Plastic Jugs and Bottles"
            onPress={() =>
              navigation.navigate("InstructionScreen", {
                documentName: "Plastic Jugs and Bottles",
              })
            }
          ></AppButton>
          <AppButton
            title="Plastic Wrap and Film"
            onPress={() =>
              navigation.navigate("InstructionScreen", {
                documentName: "Plastic Wrap and Film",
              })
            }
          ></AppButton>
        </View>
      </View>
    </Screen>
  );
}

const styles = StyleSheet.create({
  buttonContainer: {
    width: "90%",
  },
  container: {
    flex: 1,
    alignItems: "center",
    backgroundColor: colors.white,
  },
  heading: {
    marginTop: 20,
    marginBottom: 20,
    paddingLeft: 2,
    paddingRight: 2,
  },
});
export default PlasticScreen;
