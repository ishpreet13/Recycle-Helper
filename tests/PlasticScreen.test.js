import "react-native";
import React from "react";
import renderer, { act } from "react-test-renderer";
import { render, fireEvent, waitFor } from "@testing-library/react-native";

import PlasticScreen from "../app/screens/SubCategoryScreens/PlasticScreen";

act(() => {
  describe("Testing PlasticScreen ", () => {
    let wrapper;
    it("should render", async () => {
      act(() => {
        wrapper = renderer.create(<PlasticScreen />);
      });
      expect(wrapper.toJSON()).toBeTruthy();
    });
  });
});

act(() => {
  it("PlasticScreen should render correctly", async () => {
    let tree;
    act(() => {
      tree = renderer.create(<PlasticScreen />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe("PlasticScreen ", () => {
  it("navigates on button press", () => {
    const navigate = jest.fn();
    const { getByText } = render(<PlasticScreen navigation={{ navigate }} />);
    fireEvent.press(getByText("Packing Peanuts"));
    expect(navigate).toHaveBeenCalledWith("InstructionScreen", {
      documentName: "Packing Peanuts",
    });
  });
});

describe("PlasticScreen ", () => {
  it("navigates on button press", () => {
    const navigate = jest.fn();
    const { getByText } = render(<PlasticScreen navigation={{ navigate }} />);
    fireEvent.press(getByText("Plastic Containers"));
    expect(navigate).toHaveBeenCalledWith("InstructionScreen", {
      documentName: "Plastic Container",
    });
  });
});

describe("PlasticScreen ", () => {
  it("navigates on button press", () => {
    const navigate = jest.fn();
    const { getByText } = render(<PlasticScreen navigation={{ navigate }} />);
    fireEvent.press(getByText("Plastic Bags"));
    expect(navigate).toHaveBeenCalledWith("InstructionScreen", {
      documentName: "Plastic Bags",
    });
  });
});

describe("PlasticScreen ", () => {
  it("navigates on button press", () => {
    const navigate = jest.fn();
    const { getByText } = render(<PlasticScreen navigation={{ navigate }} />);
    fireEvent.press(getByText("Plastic Cap and Lids"));
    expect(navigate).toHaveBeenCalledWith("InstructionScreen", {
      documentName: "Plastic Cap and Lids",
    });
  });
});

describe("PlasticScreen ", () => {
  it("navigates on button press", () => {
    const navigate = jest.fn();
    const { getByText } = render(<PlasticScreen navigation={{ navigate }} />);
    fireEvent.press(getByText("Plastic Jugs and Bottles"));
    expect(navigate).toHaveBeenCalledWith("InstructionScreen", {
      documentName: "Plastic Jugs and Bottles",
    });
  });
});

describe("PlasticScreen ", () => {
  it("navigates on button press", () => {
    const navigate = jest.fn();
    const { getByText } = render(<PlasticScreen navigation={{ navigate }} />);
    fireEvent.press(getByText("Plastic Wrap and Film"));
    expect(navigate).toHaveBeenCalledWith("InstructionScreen", {
      documentName: "Plastic Wrap and Film",
    });
  });
});
