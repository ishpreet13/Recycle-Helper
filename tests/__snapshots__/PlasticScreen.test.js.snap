// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlasticScreen should render correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "flex": 1,
        "paddingTop": 20,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 20,
          "marginTop": 20,
          "paddingLeft": 2,
          "paddingRight": 2,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "yellowgreen",
            "fontFamily": "Avenir",
            "fontSize": 22,
            "fontWeight": "bold",
            "margin": 10,
            "textAlign": "center",
          }
        }
      >
        Select Plastic Category for Recycle Instructions
      </Text>
    </View>
    <View
      style={
        Object {
          "width": "90%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "yellowgreen",
            "borderColor": "#fff",
            "borderRadius": 25,
            "justifyContent": "center",
            "marginVertical": 10,
            "opacity": 1,
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 20,
                "fontWeight": "bold",
                "textAlign": "center",
                "textTransform": "uppercase",
              },
              Object {
                "fontSize": 20,
              },
            ]
          }
        >
          Packing Peanuts
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "yellowgreen",
            "borderColor": "#fff",
            "borderRadius": 25,
            "justifyContent": "center",
            "marginVertical": 10,
            "opacity": 1,
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 20,
                "fontWeight": "bold",
                "textAlign": "center",
                "textTransform": "uppercase",
              },
              Object {
                "fontSize": 20,
              },
            ]
          }
        >
          Plastic Containers
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "yellowgreen",
            "borderColor": "#fff",
            "borderRadius": 25,
            "justifyContent": "center",
            "marginVertical": 10,
            "opacity": 1,
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 20,
                "fontWeight": "bold",
                "textAlign": "center",
                "textTransform": "uppercase",
              },
              Object {
                "fontSize": 20,
              },
            ]
          }
        >
          Plastic Bags
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "yellowgreen",
            "borderColor": "#fff",
            "borderRadius": 25,
            "justifyContent": "center",
            "marginVertical": 10,
            "opacity": 1,
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 20,
                "fontWeight": "bold",
                "textAlign": "center",
                "textTransform": "uppercase",
              },
              Object {
                "fontSize": 20,
              },
            ]
          }
        >
          Plastic Cap and Lids
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "yellowgreen",
            "borderColor": "#fff",
            "borderRadius": 25,
            "justifyContent": "center",
            "marginVertical": 10,
            "opacity": 1,
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 20,
                "fontWeight": "bold",
                "textAlign": "center",
                "textTransform": "uppercase",
              },
              Object {
                "fontSize": 20,
              },
            ]
          }
        >
          Plastic Jugs and Bottles
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "yellowgreen",
            "borderColor": "#fff",
            "borderRadius": 25,
            "justifyContent": "center",
            "marginVertical": 10,
            "opacity": 1,
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 20,
                "fontWeight": "bold",
                "textAlign": "center",
                "textTransform": "uppercase",
              },
              Object {
                "fontSize": 20,
              },
            ]
          }
        >
          Plastic Wrap and Film
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
